### PR TITLE
refactor: Create Item Category overhaul mod for future integration

### PR DESCRIPTION
**Pull request by RoyalFox2140**
### Required

- [x] I wrote the PR title in [conventional commit format](https://docs.cataclysmbn.org/en/contribute/changelog_guidelines/).
- [x] I ran the [code formatter](https://docs.cataclysmbn.org/en/contribute/contributing/#code-style).
- [x] I linked any relevant issues using [github keyword syntax](https://docs.cataclysmbn.org/en/contribute/contributing/#pull-request-notes) so it can be closed automatically.
- [x] I have [committed my changes to new branch that isn't `main`](https://docs.cataclysmbn.org/en/contribute/contributing/#make-your-changes) so it won't cause conflict when updating `main` branch later.
## Purpose of change

Our item categories have always been hellishly bad. Ovens, crowbars, lockpicks, hammers, toolboxes, fridges, laptops, cooking pots, sewing kits, and so on are all tools. Many of the items are all under spare parts or under other. It's incredibly hard to sort efficiently. This adds a TON OF CATEGORIES. The idea is to maximize sorting, maintain mod compatibility and minimizing effort required to navigate inventory.

## Describe the solution

All categories defined in JSON, with new zones, with no zones deleted. IT IS REQUIRED NOT TO DELETE CATEGORIES! ALL MODS WOULD BREAK! This is a Default mod but this is not going to be a mod for more than a few weeks. It is planned to keep a long feedback period where we iron out all problems and implement feedback, then to merge all relevant changes into the main game. This will become a permanent feature once it's had long enough time to bake

Tools are broken into purpose. Entry, Farming, Cooking, Workshop, Chemistry, and Electronics. Everything else remains in default tools, now renamed to "unsorted tools"

Deployables is its own category, we had so many. This includes traps, furniture, vehicles, anything you deploy.

Cooking Ingredients was added, its stuff you cook with but can't eat (or really shouldnt). In the future we need to migrate all spices to this but I can't edit abstracts in a mod, it must be mainline. For now all pickling recipes are an ingredient and so is rock salt.

Other and Spare Parts is broken up into scrap categories for Metal, Electronics, Fabric, Wood, Plastic, Ceramics, Glass, and Valuables. Additionally new categories for soil and rocks. In the case of soil it's because it's a massive item you get massive quantities of and it's an item with no business being sorted on top of any other item which hides it. For rocks it handles a lot of minerals, rocks, concrete, mortar, etc.

## Describe alternatives you've considered

Theres a hundred different ideas of how categories should be. The whole community is here to give alternatives.

## Testing
Some screenshots. More available on request but I will refrain from spamming the PR, talk to me on Discord. This needs playtesting. We all need to playtest it. This was primarily done manually because no script can make decisions on what items need updating and to which category. We need a serious audit of my work to ensure no manual errors were made.

image
image
image
image

## Additional context

In Hell we live. This was 502 items copy-from'd and updated.

### CODE REVIEW

**Comment by RoyalFox2140** [comment]
Ran the JSON formatter bundled in the game on this. I figured it worked but I guess it was too large. RIP, the tree tolls for us all.

**Comment by RoyalFox2140** [comment]
Final tally of 569 copy-froms. I need a drink.

File: data/mods/Item_Category_Overhaul/categories.json [hunk 1]
```diff
@@ -0,0 +1,265 @@
+[
+  {
+    "id": "tools",
+    "type": "ITEM_CATEGORY",
+    "name": { "str": "UNSORTED TOOLS" },
```
Comment (chaosvolt) [inline data/mods/Item_Category_Overhaul/categories.json:5]: Might work better than unsorted? Not sure.
```suggestion
    "name": { "str": "OTHER TOOLS" },
```

**Comment by chaosvolt** [comment]
I only just now realized this is a mod which is...uh. I'm not sure why it can't just be in vanilla given this won't play that nice with any other mod ever without dedicated patchmods?

**Comment by RoyalFox2140** [comment]
To summarize why it was suggested to be a default mod, @KheirFerrum and I discussed it, along with me discussing it with others.

1. A massive overhaul this big requires a lengthy period of community feedback. It WILL become default behavior.
2. Doing all this all at once would disrupt everyone's game and require the whole game to be updated all at once.
3. Mods are already by default compatible. We are not removing any categories and never will.

**Comment by chaosvolt** [comment]
>     1. A massive overhaul this big requires a lengthy period of community feedback. It WILL become default behavior.

Ah, in that case nice.

File: data/mods/Item_Category_Overhaul/categories.json [hunk 2]
```diff
@@ -0,0 +1,265 @@
...
+    "type": "ITEM_CATEGORY",
+    "name": { "str": "ARTIFACTS" },
+    "zone": "LOOT_ARTIFACTS",
+    "//": "Please don't touch the sort rank, I did this so these are always on bottom.",
+    "sort_rank": 49
+  },
+  {
+    "id": "maps",
+    "type": "ITEM_CATEGORY",
+    "name": { "str": "INFORMATION" },
```
Comment (chaosvolt) [inline data/mods/Item_Category_Overhaul/categories.json:260]:
```suggestion
    "name": { "str": "INFORMATION" },
```
Given the weirdness with hardcoded city names on any item in this category gaining city names, may want to wait and mess with this after, and let the other items being put in this category stay in "other" for now.

**Review by chaosvolt — APPROVED** [review]
Looks okay at first glance and all tests pass. I suspect there will need to be more quibbling over whether some of these categories need to exist, some things I could note:
1. Cooking and chemistry tools may be seen as redundant since one skill covers all of it and you may often need boiling or distillation for chemistry tasks too (plus pots are outright sources of chemistry quality level 1), but at the same time there are enough tools that're useful for advanced chemsitry only and not general cooking that a split based on that one-way relationship still seems reasonable.
2. Soil is going to be weird category because on the one hand there's like only one item that's ever really going to be in that category, but on the other hand making sure it doesn't get lumped in with other items because it's SO FUCKING HUGE is basically the entire reason it was set aside to begin with, so bleh.

**Comment by RoyalFox2140** [comment]
Since Scarf's typescript ensured we have 0 errors and it's approved, I'm going to send this down the line for a period of several weeks of community testing and feedback. All changes will be done in future Pull Requests as if we keep editing this it will never get merged.

**Comment by Lamandus** [comment]
the new Zone categories are frankly not a good idea. The Zones are already very full of stuff that should be "together". Especially on phones it requires a lot of more work, it requires setting up 21 new zones.

I don't mind to give zones the possibility to be divided into tool sub categories. But leave the original "tools" category for zones as is! It works for food, too (Zone Food is used globally for any food item or drink item, but you can also use P.Food or P.Drink. If you need diversity.

grafik
another Problem is, Zones need more letters defined now.

**Pull request merged**